api/tasks.go

```diff
@@ -373,10 +373,13 @@ type Task struct {
 }
 
 func (t *Task) Canonicalize(tg *TaskGroup, job *Job) {
-	min := MinResources()
-	min.Merge(t.Resources)
-	min.Canonicalize()
-	t.Resources = min
+	if t.Resources == nil {
```

Comment:
Hm, this makes `MinResources` unused, but I think that's ok. Users of the `api` package still may find it useful. The old code was pretty weird anyway since MinResources and Canonicalize set the same fields so calling MinResources was already worthless.

You could simplify this to:

```go
if t.Resources == nil {
  t.Resources = &Resources{}
}
t.Resources.Canonicalize()
```